src/cascadia/TerminalControl/TSFInputControl.cpp

```diff
@@ -0,0 +1,380 @@
+    // Return Value:
+    // - <none>
+    void TSFInputControl::_selectionRequestedHandler(CoreTextEditContext sender, CoreTextSelectionRequestedEventArgs const& /*args*/)
+    {
+    }
+
+    // Method Description:
+    // - Handler for SelectionUpdating event by CoreEditContext responsible
+    //   for handling modifications to the range of text currently selected.
+    //   TSFInputControl doesn't currently allow selection, so nothing happens.
+    //   NOTE: Documentation says application should set its selection range accordingly
+    // Arguments:
+    // - sender: CoreTextEditContext sending the request. Not used in method.
+    // - args: CoreTextSelectionUpdatingEventArgs for providing data for the SelectionUpdating event. Not used in method.
+    // Return Value:
+    // - <none>
+    void TSFInputControl::_selectionUpdatingHandler(CoreTextEditContext sender, CoreTextSelectionUpdatingEventArgs const& /*args*/)
+    {
+    }
+
+    // Method Description:
+    // - Handler for TextUpdating event by CoreEditContext responsible
+    //   for handling text updates.
+    // Arguments:
+    // - sender: CoreTextEditContext sending the request. Not used in method.
+    // - args: CoreTextTextUpdatingEventArgs contains new text to update buffer with.
+    // Return Value:
+    // - <none>
+    void TSFInputControl::_textUpdatingHandler(CoreTextEditContext sender, CoreTextTextUpdatingEventArgs const& args)
+    {
+        const auto text = args.Text();
+        const auto range = args.Range();
+
+        try
+        {
+            _inputBuffer = _inputBuffer.replace(
+                range.StartCaretPosition,
+                static_cast<size_t>(range.EndCaretPosition) - static_cast<size_t>(range.StartCaretPosition),
+                text.c_str());
+
+            _textBlock.Text(_inputBuffer);
+
+            // Notify the TSF that the update succeeded
+            args.Result(CoreTextTextUpdatingResult::Succeeded);
+        }
+        catch (...)
+        {
+            LOG_CAUGHT_EXCEPTION();
+
+            // indicate
+            args.Result(CoreTextTextUpdatingResult::Failed);
+        }
+    }
+
+    // Method Description:
+    // - Handler for FormatUpdating event by CoreEditContext responsible
+    //   for handling different format updates for a particular range of text.
+    //   TSFInputControl doesn't do anything with this event.
+    // Arguments:
+    // - sender: CoreTextEditContext sending the request. Not used in method.
+    // - args: CoreTextFormatUpdatingEventArgs Provides data for the FormatUpdating event. Not used in method.
+    // Return Value:
+    // - <none>
+    void TSFInputControl::_formatUpdatingHandler(CoreTextEditContext sender, CoreTextFormatUpdatingEventArgs const& /*args*/)
+    {
+    }
+
+    DEFINE_EVENT_WITH_TYPED_EVENT_HANDLER(TSFInputControl, CurrentCursorPosition, _currentCursorPositionHandlers, TerminalControl::TSFInputControl, TerminalControl::CursorPositionEventArgs);

```

Comment:
These can use the new `TYPED_EVENT` helper and be removed from the cpp file. It's a new version of the old bipartite macro!

Reply:
Nice, will address in next push!